extensions/amp-a4a/0.1/template-validator.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdTemplateHelper} from '../../amp-a4a/0.1/amp-ad-template-helper';
+import {Services} from '../../../src/services';
+import {Validator, ValidatorResult} from './amp-ad-type-defs';
+import {getAmpAdMetadata} from './amp-ad-utils';
+import {pushIfNotExist} from '../../../src/utils/array';
+import {tryParseJson} from '../../../src/json';
+import {utf8Decode} from '../../../src/utils/bytes';
+
+/** @const {string} */
+export const AMP_TEMPLATED_CREATIVE_HEADER_NAME = 'AMP-template-amp-creative';
```

Comment:
As stated in the design doc, we'd like to rename this to `AMP-Ad-Template-Extension`. Meantime, please make sure the we don't break AdZerk. So we might end of having both, until AdZerk getting migrated over.